server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -311,6 +312,24 @@ public NodeEnvironment(Settings settings, Environment environment, Consumer<Stri
 
             applySegmentInfosTrace(settings);
             assertCanWrite();
+
+            // backported from 7.0, but turned into warnings.
+            if (DiscoveryNode.isDataNode(settings) == false) {
+                if (DiscoveryNode.isMasterNode(settings) == false) {
+                    try {
+                        ensureNoIndexMetaData(nodePaths);
+                    } catch (IllegalStateException e) {
+                        logger.warn(e.getMessage() + ", this should be cleaned up (will refuse to start in 7.0). Beware of data-loss.");
```

Comment:
I think we should use the deprecation logger here (see `DeprecationLogger` class). That one will log to a different log file. This will help users in finding all things to address before upgrading

Reply:
I'm not sure about the "Beware of data-loss." part of the message. That's not really actionable. Perhaps it could say something along the lines of "create a backup copy before removing."